### PR TITLE
handle enter key events in the ha-switch

### PR DESCRIPTION
**Pull request by iantrich**

### CODE REVIEW

File: src/components/ha-switch.ts [hunk 1]
```diff
@@ -21,6 +21,12 @@ export class HaSwitch extends MwcSwitch {
       "slotted",
       Boolean(this._slot.assignedNodes().length)
     );
+
+    this.addEventListener("keyup", (ev) => {
```
Comment (bramkragten) [inline src/components/ha-switch.ts]:
```suggestion
    this.addEventListener("keyup", (ev: KeyboardEvent) => {
```

**Review by bramkragten — APPROVED** [review]

**Comment by bramkragten** [comment]
I see they do support the spacebar, it is the same as a regular checkbox, because a enter would submit the form... Wonder if we should add this...

**Comment by iantrich** [comment]
is that the standard? I have no idea

**Comment by iantrich** [comment]
@bramkragten 
> I see they do support the spacebar, it is the same as a regular checkbox, because a enter would submit the form... Wonder if we should add this...

https://webaim.org/techniques/keyboard/

> Testing with a keyboard is an essential part of any accessibility evaluation. The basics of keyboard testing are simple—the Tab key can be used to navigate through links and form controls, Enter (and sometimes Spacebar) selects an element, and the arrow keys are sometimes used for other navigation.

**Comment by iantrich** [comment]
@bramkragten I think we should move forward with adding enter support for the switch based on the docs I linked

**Comment by bramkragten** [comment]
Reading that link I think spacebar makes the most sense for this component, as a switch would be comparable with a `checkbox`? So I think we should close this.

**Comment by iantrich** [comment]
you're absolutely right, I must have had toggle-**button** stuck in my head :stuck_out_tongue_winking_eye:

**Pull request closed without merging**